### PR TITLE
Move documentation from implementations to declarations #2735.

### PR DESCRIPTION
**Pull request by Sid672**
Moved the documentation comments from town.cpp to town.h .
Please review it and give your valuable feedback.

### CODE REVIEW

**Review by ephphatha — CHANGES_REQUESTED** [review]
These are doxygen comments, they need to be located in the source on the lines immediately above a function declaration.

This means that only the functions already declared in the header need to have the comment moved to the header.

**Comment by ephphatha** [comment]
https://github.com/diasurgical/devilutionX/pull/2740/files provides a good example of how this should be done.

File: Source/town.h [hunk 1]
```diff
@@ -2,7 +2,40 @@
...
+ *
+ * Function: TownOpenHive().
+ * @brief Update the map to show the open hive
+ *
+ * Function: TownOpenGrave().
+ * @brief Update the map to show the open grave
+ *
+ * Function: CreateTown.
+ * @brief Initialize town level
+ * @param entry Methode of entry
```
Comment (AJenbo) [inline Source/town.h:36]: Each comment should still be right above the function that it documents. To goal here is to move the documentation from the implementation to the decleration (`void CreateTown(lvl_entry entry);`), for this block that would be line 48.

File: Source/town.cpp [hunk 2]
```diff
@@ -130,9 +114,6 @@ void TownCloseHive()
 	SetDungeonMicros();
 }
 
-/**
- * @brief Update the map to show the closed grave
- */
 void TownCloseGrave()
```
Comment (AJenbo) [inline Source/town.cpp:117]: TownCloseGrave() is a function local to the cpp unit, it is not declared elsewhere and so it's documentation should not be moved.

**Review by AJenbo — CHANGES_REQUESTED** [review]
Only move the documentation of functions that are publicly declared in the header.
The documentation still to be next to the relevant function declaration so that it can be picked up correctly by tools.
See this commit for an example: https://github.com/diasurgical/devilutionX/commit/51f50570da5e169e310d07d813c09207c2285974

File: Source/town.cpp (hunk 2 above)
Comment (Sid672) [reply Source/town.cpp:117]: Okay, now it's more clear for me. Thanks.

File: Source/town.h [hunk 3]
```diff
@@ -3,15 +3,28 @@
  *
  * Interface of functionality for rendering the town, towners and calling other render routines.
  */
+
```
Comment (AJenbo) [inline Source/town.h:6]:
```suggestion
```

File: Source/town.cpp [hunk 4]
```diff
@@ -148,9 +129,6 @@ void TownCloseGrave()
 	SetDungeonMicros();
 }
 
-/**
- * @brief Initialize all of the levels data
- */
```
Comment (AJenbo) [inline Source/town.cpp:153]: Do not delete documentation from local function that do not have a header entry.

File: Source/town.h [hunk 5]
```diff
@@ -10,8 +10,20 @@
...
 void TownOpenHive();
+
+/** 
+* @brief Update the map to show the open grave
+*/
 void TownOpenGrave();
+
+/**
+* @brief Initialize town level
+* @param entry Methode of entry
```
Comment (qndel) [inline Source/town.h:25]:
```suggestion
* @param entry Method of entry

```

File: Source/town.cpp [hunk 6]
```diff
@@ -130,9 +130,9 @@ void TownCloseHive()
 	SetDungeonMicros();
 }
 
-/**
- * @brief Update the map to show the closed grave
- */
+/** 
+* @brief Update the map to show the closed grave
+*/
```
Comment (AJenbo) [inline Source/town.cpp:135]: Indentation is wrong and there is some trailing white space (see code style), please make sure that lines you did not intend to change do not show up as changed.
```suggestion
/**
 * @brief Update the map to show the closed grave
 */
```

File: Source/town.h [hunk 7]
```diff
@@ -1,17 +1,30 @@
...
 namespace devilution {
 
+/**
+* @brief Update the map to show the open hive
+*/
 void TownOpenHive();
+
+/** 
+* @brief Update the map to show the open grave
+*/
```
Comment (AJenbo) [inline Source/town.h:21]: Doxygen comments should have the stars aligned:
```suggestion
/** 
 * @brief Update the map to show the open grave
 */
```

Comment (Sid672) [reply Source/town.h:21]: ok

**Comment by qndel** [comment]
https://github.com/diasurgical/devilutionX/pull/2767/files
You can literally see all your changes there, half of them are whitespaces / newlines/ whatever that shouldn't be there.

**Comment by Sid672** [comment]
> https://github.com/diasurgical/devilutionX/pull/2767/files
> You can literally see all your changes there, half of them are whitespaces / newlines/ whatever that shouldn't be there.

I am sorry for not doing it properly but can you please tell me which whitespaces/ newlines you want to remove.
So that I can update it in a better way.

**Comment by qndel** [comment]
> > https://github.com/diasurgical/devilutionX/pull/2767/files
> > You can literally see all your changes there, half of them are whitespaces / newlines/ whatever that shouldn't be there.
> 
> I am sorry for not doing it properly but can you please tell me which whitespaces/ newlines you want to remove.
> So that I can update it in a better way.

Not really, I'm not here to babysit you, if it's not obvious from looking at the differences you introduced, there's nothing to do here for me 🤷 you've been given all necessary instructions already.

File: Source/town.h [hunk 8]
```diff
@@ -3,15 +3,28 @@
  *
  * Interface of functionality for rendering the town, towners and calling other render routines.
  */
+
```
Comment (AJenbo) [inline Source/town.h:6]: There should not be new changes in lines you did not intend to change:
```suggestion
```

File: Source/town.cpp [hunk 9]
```diff
@@ -150,7 +150,7 @@ void TownCloseGrave()
 
 /**
  * @brief Initialize all of the levels data
- */
+ */
```
Comment (AJenbo) [inline Source/town.cpp:153]:
```suggestion
 */
```

File: Source/town.cpp [hunk 10]
```diff
@@ -130,7 +130,7 @@ void TownCloseHive()
 	SetDungeonMicros();
 }
 
-/**
+/**
```
Comment (AJenbo) [inline Source/town.cpp:133]:
```suggestion
/**
```

File: Source/town.cpp [hunk 11]
```diff
@@ -76,7 +76,7 @@ void FillTile(int xx, int yy, int t)
 	dPiece[xx + 1][yy + 1] = SDL_SwapLE16(mega.micro4) + 1;
 }
 
-/**
+/**
```
Comment (AJenbo) [inline Source/town.cpp:79]:
```suggestion
/**
```

File: Source/town.cpp [hunk 12]
```diff
@@ -17,7 +17,7 @@ namespace devilution {
 
 namespace {
 
-/**
+/**
```
Comment (AJenbo) [inline Source/town.cpp:20]:
```suggestion
/**
```

**Review by AJenbo — CHANGES_REQUESTED** [review]
The only changes should be the 3 comments you moved, the rest of the files should stay unchanged.

File: Source/town.h (hunk 8 above)
Comment (Sid672) [reply Source/town.h:6]: Ok.

**Comment by AJenbo** [comment]
You don't need to move the coments, you need to remove the spaces that you added by mistake. Please move the comments back again.

**Pull request merged**